### PR TITLE
init: use *os.File for passed file descriptors

### PR DESCRIPTION
**Pull request by cyphar**
While it doesn't make much of a practical difference, it seems far more reasonable to use os.NewFile to wrap all of our passed file descriptors to make sure they're tracked by the Go runtime and that we don't double-close them.

Signed-off-by: Aleksa Sarai <cyphar@cyphar.com>

### CODE REVIEW

File: libcontainer/utils/utils_unix.go [hunk 1]
```diff
@@ -202,3 +202,8 @@ func ProcThreadSelf(subpath string) (string, ProcThreadSelfCloser) {
 	}
 	return threadSelf + subpath, runtime.UnlockOSThread
 }
+
+// ProcThreadSelfFd is shorthand for ProcThreadSelf(fmt.Sprintf("fd/%d", fd)).
```
Comment (AkihiroSuda) [inline libcontainer/utils/utils_unix.go:206]: The comment is out of sync with the actual implementation

Comment (cyphar) [reply libcontainer/utils/utils_unix.go:206]: It isn't the same as the actual implementation, but I thought the description being easily understandable was more important than matching the implementation exactly.

But I'll reword this to just explain what it does.

**Review by AkihiroSuda — APPROVED** [review]

**Comment by lifubang** [comment]
I have updated #4173 , but after I see this PR, I think these two PRs are related, could you mind carry #4173 in here?

**Review by kolyshkin — APPROVED** [review]
LGTM
